iothub/device/src/InternalClient.cs

```diff
@@ -57,58 +50,40 @@ internal class InternalClient : IDisposable
+                Logging.Enter(this, iotHubClientOptions?.TransportSettings, nameof(InternalClient) + "_ctor");
 
-            IotHubConnectionCredentials = iotHubConnectionCredentials;
-            _clientOptions = iotHubClientOptions;
-
-            if (!iotHubClientOptions.ModelId.IsNullOrWhiteSpace()
-                && iotHubClientOptions.TransportSettings is IotHubClientHttpSettings)
+            // Make sure client options is initialized.
+            if (iotHubClientOptions == default)
             {
-                throw new InvalidOperationException("Plug and Play is not supported over the HTTP transport.");
+                iotHubClientOptions = new();
             }
 
-            var pipelineContext = new PipelineContext
+            IotHubConnectionCredentials = iotHubConnectionCredentials;
+            _clientOptions = iotHubClientOptions;
+
+            _pipelineContext = new PipelineContext
             {
-                IotHubConnectionCredentials = iotHubConnectionCredentials,
-                ProductInfo = iotHubClientOptions.ProductInfo,
-                IotHubClientTransportSettings = iotHubClientOptions.TransportSettings,
-                ModelId = iotHubClientOptions.ModelId,
+                IotHubConnectionCredentials = IotHubConnectionCredentials,
+                ProductInfo = _clientOptions.ProductInfo,
+                IotHubClientTransportSettings = _clientOptions.TransportSettings,
+                ModelId = _clientOptions.ModelId,
                 MethodCallback = OnMethodCalledAsync,
                 DesiredPropertyUpdateCallback = OnDesiredStatePatchReceived,
                 ConnectionStatusChangeHandler = OnConnectionStatusChanged,
-                ModuleEventCallback = OnModuleEventMessageReceivedAsync,
-                DeviceEventCallback = OnDeviceMessageReceivedAsync,
             };
 
-            pipelineBuilder ??= BuildPipeline();
-            IDelegatingHandler innerHandler = pipelineBuilder.Build(pipelineContext);
-
-            if (Logging.IsEnabled)
-                Logging.Associate(this, innerHandler, nameof(InternalClient));
-
-            InnerHandler = innerHandler;
-
             if (Logging.IsEnabled)
-                Logging.Associate(this, iotHubClientOptions.TransportSettings, nameof(InternalClient));
-
-            _fileUploadHttpTransportHandler = new HttpTransportHandler(pipelineContext, iotHubClientOptions.FileUploadTransportSettings);
-
-            if (Logging.IsEnabled)
-                Logging.Exit(this, iotHubClientOptions.TransportSettings, pipelineBuilder, nameof(InternalClient) + "_ctor");
+                Logging.Exit(this, _clientOptions.TransportSettings, nameof(InternalClient) + "_ctor");
         }
 
         /// <summary>
         /// Diagnostic sampling percentage value, [0-100];
-        /// A value of 0 means no message will carry on diagnostics info.
+        /// 0 means no message will carry on diagnostics info
```

Comment:
I don't think this change is better.

Reply:
Ah, I copied this over from the public class. I agree that this change isn't better than what we previously had on this class. I'll revert it.